packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -445,6 +446,25 @@ void main() {
     expect(cacheDill, isNot(exists));
   }));
 
+  test('ResidentRunner can run source generation', () => testbed.run(() async {
+    final FakeProcessManager processManager = globals.processManager as FakeProcessManager;
+    processManager.addCommand(FakeCommand(
+      command: <String>[
+        globals.artifacts.getArtifactPath(Artifact.engineDartBinary),
+        globals.fs.path.join(Cache.flutterRoot, 'dev', 'tools', 'localization', 'bin', 'gen_l10n.dart'),
+      ],
+    ));
+    globals.fs.file(globals.fs.path.join('lib', 'l10n', 'foo.arb'))
+      .createSync(recursive: true);
+    globals.fs.file('l10n.yaml').createSync();
+
+    await residentRunner.runSourceGenerators();
+
+    expect(testLogger.errorText, isEmpty);
+  }, overrides: <Type, Generator>{
```

Comment:
What does this override do?

Reply:
Its a bit of hack for https://github.com/flutter/flutter/issues/47161. Basically we need to inject this into the context. Refactoring the resident runner depends on refactoring all of the code it depends on